pkg/services/sqlstore/annotations_cleanup.go

```diff
@@ -0,0 +1,49 @@
+package sqlstore
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana/pkg/bus"
+	m "github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/util/errutil"
+)
+
+func init() {
+	bus.AddHandler("sql", deleteExpiredAnnotations)
+}
+
+const MAX_EXPIRED_ANNOTATIONS_TO_DELETE = 900
+
+func deleteExpiredAnnotations(cmd *m.DeleteExpiredAnnotationsCommand) error {
+	return inTransaction(func(sess *DBSession) error {
+		historyTimeStamp := (time.Now().Unix() - int64(cmd.DaysToKeep*86400)) * 1000
+
+		query := "SELECT id FROM annotation WHERE created <= ? ORDER BY id LIMIT ?"
+		var annotationIdsToDelete []interface{}
+		if err := sess.SQL(query, historyTimeStamp, MAX_EXPIRED_ANNOTATIONS_TO_DELETE).Find(&annotationIdsToDelete); err != nil {
+			return errutil.Wrapf(err, "failed to query for expired annotations")
+		}
+		if len(annotationIdsToDelete) == 0 {
+			return nil
+		}
+
+		query = fmt.Sprintf("DELETE FROM annotation_tag WHERE annotation_id IN (?%s)", strings.Repeat(",?",
```

Comment:
Are we comfortable/sure this will not lock the table and interfere with alerting?

Reply:
This was build with approach identical to how cleaning of **dashboard_version** table is done, which is currently the biggest in our internal system. Cleaning annotations is never being an issue for our MySQL server.